resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -338,13 +338,14 @@ private[spark] class Client(
       throw new IllegalArgumentException(s"Required executor memory ($executorMemory" +
         s"+$executorMemoryOverhead MB) is above the max threshold ($maxMem MB) of this cluster! " +
         "Please check the values of 'yarn.scheduler.maximum-allocation-mb' and/or " +
-        "'yarn.nodemanager.resource.memory-mb'.")
+        "'yarn.nodemanager.resource.memory-mb and increase the memory appropriately.")
```

Comment:
What does this add?

Reply:
As mentioned in the JIRA even though the memory defined in the yarn.nodemanager.resource.memory-mb parameter is less eg: 
yarn.scheduler.maximum-allocation-mb =15g and yarn.nodemanager.resource.memory-mb =8g
Launch spark-shell --master yarn --conf spark.yarn.am.memory=10g
we are getting below error 
java.lang.IllegalArgumentException: Required AM memory (10240+1024 MB) is above the max threshold (8096 MB) of this cluster! Please increase the value of 'yarn.scheduler.maximum-allocation-mb'.
This message is very confusing to a user since spark indicate  to increase the arn.scheduler.maximum-allocation-mb which is actually more than 10G as per this scenario whereas the issue is with yarn.nodemanager.resource.memory-mb,

Reply:
Both settings may have to change. But your text change doesn't mention anything new. Why does this help? I also don't think that the current behavior is wrong

Reply:
Where as  in the  executor memory validation check we are displaying the proper message considering both   yarn.nodemanager.resource.memory-mb  and yarn.scheduler.maximum-allocation-mb
```
    if (executorMem > maxMem) {
      throw new IllegalArgumentException(s"Required executor memory ($executorMemory" +
        s"+$executorMemoryOverhead MB) is above the max threshold ($maxMem MB) of this cluster! " +
        "Please check the values of 'yarn.scheduler.maximum-allocation-mb' and/or " +
        "'yarn.nodemanager.resource.memory-mb and increase the memory appropriately.")
    }
```
so same message i think is required for am memory validation as well

Reply:
Please let me know for any suggestions. Thanks

Reply:
Yes both settings has to be changed, but for the AM container memory allocation validation only yarn.scheduler.maximum-allocation-mb parameter is mentioned message does not indicate to increase yarn.nodemanager.resource.memory-mb,  in my change i updated this message by including  "yarn.nodemanager.resource.memory-mb" paramter

Reply:
I understand the change below, that's probably fine. The change above doesn't seem to add anything. You want to tell the user to check both settings in both cases.

Reply:
just trying to make the messages more informative by mentioning to increase the memory appropriately as per the business use-case .i shall remove the part if it doesn't make much difference to the information we are trying to convey.

Reply:
Sure, though the solution may be to reduce the app memory usage rather than increase the YARN limits (which only admins can do). I think that's why the message more neutrally suggests that these are the limits that the app is hitting.

Reply:
Got your point, i will update as per the suggestion. Thanks for the feedback Sean.